tibanna/ec2_utils.py

```diff
@@ -549,31 +549,48 @@ def launch_and_get_instance_id(self):
                 return instance_id
             
             elif 'Errors' in fleet_result and len(fleet_result['Errors']) > 0:
-                error_code = fleet_result['Errors'][0]['ErrorCode']
-                error_msg = fleet_result['Errors'][0]['ErrorMessage']
+
+                error_codes = list(map(lambda err: err['ErrorCode'], fleet_result['Errors']))
+                error_msgs = list(map(lambda err: err['ErrorMessage'], fleet_result['Errors']))
+
+                num_unique_errors = len(set(error_codes))
 
                 self.delete_fleet(fleet_result['FleetId'])
 
-                if ('InsufficientInstanceCapacity' in error_code or 'InstanceLimitExceeded' in error_code   
-                        or 'is not supported in your requested Availability Zone' in error_msg            
-                        or 'UnfulfillableCapacity' in error_code):
-                    
+                if 'InvalidLaunchTemplate' in error_codes and invalid_launch_template_retries < 5:
+                    invalid_launch_template_retries += 1
+                    logger.info(f"LaunchTemplate not found. Retry #{invalid_launch_template_retries}")
+                    # continue without creating a new launch template
+                    continue
+
+                elif 'InvalidLaunchTemplate' in error_codes and invalid_launch_template_retries >= 5:
+                    self.delete_launch_template()
+                    raise Exception(f"InvalidLaunchTemplate. Result from create_fleet command: {json.dumps(fleet_result)}")
+
+                elif num_unique_errors == 1 and 'InvalidFleetConfiguration' in error_codes:
+                    # This error code includes the "Your requested instance type (xxx) is not supported in your requested Availability Zone (xxx)" error
+                    # In this case there must be an issue with the general setup, otherwise we would get additional error codes, e.g., InsufficientInstanceCapacity
+                    self.delete_launch_template()
+                    raise Exception(f"Invalid fleet configuration. Result from create_fleet command: {json.dumps(fleet_result)}")
```

Comment:
Not sure I'm understanding the desired behavior here. I think a valid configuration could see only this error, and in fact may be a semi-common case in which you don't want to fail? I think you only want to fail in this case if you got invalid fleet configuration for every instance type in every available subnet. So this would fail unnecessarily quite frequently?